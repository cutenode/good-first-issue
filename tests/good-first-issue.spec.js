const { prepareEnvironment } = require("@gmrchk/cli-testing-library");

describe('CLI', () => {
  it('should run', async () => {
    const { spawn, cleanup } = await prepareEnvironment();
    const { waitForText, waitForFinish, getExitCode, pressKey } = await spawn('node', './bin/good-first-issue.js');

    await waitForText('Choose a project:');
    await pressKey('enter');
    await waitForFinish();

    expect(getExitCode()).toBe(0);

    await cleanup();
  });

  it('should run with -o', async () => {
    const { spawn, cleanup } = await prepareEnvironment();
    const { waitForText, waitForFinish, getExitCode, pressKey } = await spawn('node', './bin/good-first-issue.js -o');

    await waitForText('Choose a project:');
    await pressKey('enter');
    await waitForFinish();

    expect(getExitCode()).toBe(0);

    await cleanup();
  })

  it('prints an object with the correct shape', async () => {
    const { spawn, cleanup } = await prepareEnvironment();
    const { waitForText, waitForFinish, getExitCode, getStdout, pressKey } = await spawn('node', './bin/good-first-issue.js');

    await waitForText('Choose a project:');
    await pressKey('enter');
    await waitForFinish();

    expect(getExitCode()).toBe(0);
    expect(getStdout()).toMatchSnapshot();

    await cleanup();
  })
});
