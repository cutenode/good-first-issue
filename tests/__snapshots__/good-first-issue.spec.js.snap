// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CLI returns an object with the correct shape 1`] = `
Array [
  "? Choose a project: (Use arrow keys)",
  "❯ Apollo - A community building flexible open source tools for GraphQL.",
  "Babel - Babel is a compiler for writing next generation JavaScript.",
  "Create React App - Set up a modern web app by running one command.",
  "Firefox Debugger - The Firefox debugger that works anywhere.",
  "Docusaurus - Easy to maintain open source documentation websites.",
  "Docz - It has never been so easy to document your things!",
  "EasyGraphQL - EasyGraphQL is a group of open source tools, with the main focus",
  "(Move up and down to reveal more choices)",
  "? Choose a project: Apollo - A community building flexible open source tools for",
  "GraphQL.",
  "╭────────────────────────────────────────────────────────────────────╮",
  "│                                                                    │",
  "│   Good First Issue in Apollo:                                      │",
  "│                                                                    │",
  "│     - Title: Cannot set telemetry.tracing.otlp.grpc.metadata       │",
  "│     - Repository: apollographql/router                             │",
  "│     - Issue: #1166                                                 │",
  "│     - Status: open                                                 │",
  "│     - Assigned to: unassigned!                                     │",
  "│                                                                    │",
  "│   Start now: https://github.com/apollographql/router/issues/1166   │",
  "│                                                                    │",
  "╰────────────────────────────────────────────────────────────────────╯",
]
`;
